urllib3/response.py

```diff
@@ -351,3 +363,71 @@ def readinto(self, b):
         else:
             b[:len(temp)] = temp
             return len(temp)
+
+    def _read_chunked(self, amt=None):
+        assert self.chunked
+        chunk_left = self.chunk_left
+        value = []
+        while True:
+            if chunk_left is None:
+                line = self._fp.fp.readline(amt)
+                i = line.find(';')
+                if i >= 0:
+                    line = line[:i] # strip chunk-extensions
+                try:
+                    chunk_left = int(line, 16)
+                except ValueError:
+                    # close the connection as protocol synchronisation is
+                    # probably lost
+                    self.close()
+                    raise httplib.IncompleteRead(''.join(value))
+                if chunk_left == 0:
+                    break
+            if amt is None:
+                yield self._fp._safe_read(chunk_left)
+            elif amt < chunk_left:
+                value.append(self._fp._safe_read(amt))
+                self.chunk_left = chunk_left - amt
+                yield ''.join(value)
+            elif amt == chunk_left:
+                value.append(self._fp._safe_read(amt))
+                self._fp._safe_read(2)  # toss the CRLF at the end of the chunk
+                self.chunk_left = None
+                yield ''.join(value)
+            else:
+                yield self._fp._safe_read(chunk_left)
+                amt -= chunk_left
+
+            # we read the whole chunk, get another
+            self._fp._safe_read(2)      # toss the CRLF at the end of the chunk
+            chunk_left = None
+
+        # read and discard trailer up to the CRLF terminator
```

Comment:
Another style nit: Please make all comments sentence-like for consistency (starts with a capital, ends with a period).

:)